### PR TITLE
Add melting

### PR DESCRIPTION
**Pull request by tduretz**
Add single phase melting implementation.
Two setups are added: 
- MeltingOverPressure: shows the volumetric effect of melting as in the paper of Vrijmoed et al. (2009)
image
- CrystallisationStuewe1995: shows the buffering effect of latent heat on cooling of an intrusion
CrystallisationStuewe1995

### CODE REVIEW

**Comment by lcandiot** [comment]
Looks great! Have you tested melting in a Rifting Setup already?

File: MDLIB/HDF5Output.c
Comment (lcandiot) [inline MDLIB/HDF5Output.c:1]: Does `AddFieldToGroup(... , "Centers", "phi", ...)` not clash with storing the friction angle if strain softening is activated?

**Comment by tduretz** [comment]
yes, just did, see RiftingMelting setup  ;)

image

**Comment by lcandiot** [comment]
> yes, just did, see RiftingMelting setup ;)
> 
> image

Wow, there is quite some melt generation at the base of the crust. Is this to be expected? Looks really cool :)

**Comment by tduretz** [comment]
yes, that's because the Moho it a 700 C

File: MDLIB/HDF5Output.c
Comment (tduretz) [reply MDLIB/HDF5Output.c:1]: I don't think so. The friction angle is called "friction"
`AddFieldToGroup( FileName, "Centers" , "friction", 'f', (model.Nx-1)*(model.Nz-1), Cfriction, 1 );`
Or do you refer to something else?

**Pull request merged**